param.go

```diff
@@ -326,3 +342,52 @@ func (po paramObject) Build(c *Container) (reflect.Value, error) {
 	}
 	return dest, nil
 }
+
+// paramGroupedSlice is a param which produces a slice of values with the same
+// group name.
+type paramGroupedSlice struct {
+	// Name of the group as specified in the `group:".."` tag.
+	Group string
+
+	// Type of the slice.
+	Type reflect.Type
+}
+
+// newParamGroupedSlice builds a paramGroupedSlice from the provided type with
+// the given name.
+//
+// The type MUST be a slice type.
+func newParamGroupedSlice(group string, t reflect.Type) paramGroupedSlice {
+	return paramGroupedSlice{Group: group, Type: t}
+}
+
+func (pt paramGroupedSlice) Build(c *Container) (reflect.Value, error) {
+	k := key{group: pt.Group, t: pt.Type.Elem()}
+
+	for _, n := range c.producers[k] {
+		if err := n.Call(c); err != nil {
+			// TODO: better message
+			return _noValue, errWrapf(err, "failed to build %v", k)
```

Comment:
Is this the behavior we want? This will get difficult as modules provide e.g., middleware, that depends on types that may or may not be present. Should we default to skipping failed constructors, or should we support optional groups?

Reply:
Discussed offline with @kriskowal: Yeah, this is desirable and largely
unavoidable with a static dependency graph.